File: app/src/main/java/com/itba/runningMate/di/DependencyContainerLocator.java
package com.itba.runningMate.di;

import android.content.Context;

import androidx.annotation.VisibleForTesting;

public class DependencyContainerLocator {

    private static DependencyContainer dependencyContainer;

    private DependencyContainerLocator() {
    }

    public static DependencyContainer locateComponent(final Context context) {
        if (dependencyContainer == null) {
            dependencyContainer = new ProductionDependencyContainer(context);
        }
        return dependencyContainer;
    }

    @VisibleForTesting
    public static void setComponent(final DependencyContainer dependencyContainer) {
        DependencyContainerLocator.dependencyContainer = dependencyContainer;
    }
}

File: app/src/main/java/com/itba/runningMate/di/DependencyContainerLocator.kt
package com.itba.runningMate.di

import android.content.Context
import androidx.annotation.VisibleForTesting

object DependencyContainerLocator {

    private var dependencyContainer: DependencyContainer? = null

    @JvmStatic
    fun locateComponent(context: Context): DependencyContainer {
        if (dependencyContainer == null) {
            dependencyContainer = ProductionDependencyContainer(context)
        }
        return dependencyContainer!!
    }

    @VisibleForTesting
    fun setComponent(dependencyContainer: DependencyContainer?) {
        DependencyContainerLocator.dependencyContainer = dependencyContainer
    }
}